roofit/CMakeLists.txt

```diff
@@ -4,6 +4,10 @@
 # For the licensing terms see $ROOTSYS/LICENSE.
 # For the list of contributors see $ROOTSYS/README/CREDITS.
 
+if(ROOT_PLATFORM MATCHES "linux|macosx" AND CMAKE_SYSTEM_PROCESSOR MATCHES x86_64 AND CMAKE_CXX_COMPILER_ID MATCHES "GNU|Clang")
+  set(roofit-arch-libs ON)
```

Comment:
Hi @manolismih! 

I would prefer here not to add one more bool CMake variable, but to use one of two options: 
 - or use `if(ROOT_PLATFORM MATCHES "linux|macosx" AND CMAKE_SYSTEM_PROCESSOR MATCHES x86_64 AND CMAKE_CXX_COMPILER_ID MATCHES "GNU|Clang")` directly everywhere.
- or add a proper ROOT CMake option.

The goal is to have less 'poluted' CMake files.

cc: @Axel-Naumann

Reply:
If the variable is used at several places then a build option might make sense, else the if expression?

Reply:
At some point during the development of that PR, Manos actually got asked to make this a bool in order to not have those `if`s everywhere.
And the ROOT build option was also vetoed IIRC. :-)

Reply:
Yeah I prefer not to have yet another configuration parameter that we need to test on/off, indeed, thanks for reminding me. @oshadura can you expand on how a variable is generally worse than a config option?

Reply:
Thanks for the explanation @hageboeck. What I would like to say, at least can you add/move some note here, explaining what it is `roofit-arch-libs`? _(I was not aware of this PR so I am adding my comments now)_

Second thing, to reply Axel, looking on `roofit/CMakeLists.txt`  it is absolutelly not clear why these lines are added in this CMake file and what are their purpose.

Reply:
No problem Oksana! But @manolismih will do that. ;-)